core/src/main/java/io/questdb/TelemetryJob.java

```diff
@@ -89,47 +89,50 @@ public TelemetryJob(CairoEngine engine, @Nullable FunctionFactoryCache functionF
                 this.writer = new TableWriter(configuration, tableName);
             } catch (CairoException ex) {
                 LOG.error().$("could not open [table=").utf8(tableName).$("]").$();
-                throw ex;
+                enabled = false;
```

Comment:
when you set `enabled=false` just return out of the method to avoid tripping over second table and also peppering the rest of the code with conditionals.

When coming out of the method like that though ensure that all writers are released.

For example when first writer lock succeeds and second fails - first writer is never released.

Reply:
"return out of the method"
--> done

"For example when first writer lock succeeds and second fails - first writer is never released."
--> this is not the case, the `writer` is released.